src/dune_rules/lib.mli

```diff
@@ -57,6 +55,8 @@ type sub_system = ..
 
 type db
 
+val ppx_runtime_deps : t -> db -> t list Resolve.Memo.t
```

Comment:
The new API seems wrong to me. Once a library is resolved, it belongs to a particular database and all of its dependencies have been resolved. You should be able to delay the evaluate of ppx_runtime_deps without changing the API.